mail/mail.go

```diff
@@ -0,0 +1,65 @@
+package mail
+
+import (
+	"bytes"
+	"html/template"
+
+	"github.com/datainq/jobhub"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/gomail.v2"
+)
+
+// type SMS struct {
+// }
+
+type Mail struct {
+	From         string
+	To           string
+	Subject      string
+	Host         string
+	Port         int
+	Username     string
+	Password     string
+	TemplatePath string
+	Log          logrus.FieldLogger
+}
+
+func parseTemplate(templatePath string, data interface{}) (string, error) {
+	t := template.Must(template.ParseFiles(templatePath))
+	buf := new(bytes.Buffer)
+	if err := t.Execute(buf, data); err != nil {
+		return "", err
+	}
+	return buf.String(), nil
+}
+
+func (m *Mail) SendStatus(pipelineStatus jobhub.PipelineStatus) {
```

Comment:
This function does a lot of things. It parses, loads file, renders template and sends email. 
Let's replace `Subject` and `TemplatePath` fields by `*template.Template` and assume someone set if from outside.

```
   SubjectTmpl *template.Template
   BodyTmpl *template.Template
```

Then you can separate rendering templates from sending an email.

Also consider, that right now the `Mail` is a mix of email to be send (Subject field) and a config to be used. They are two different things, worth splitting.